pandas/core/array_algos/take.py

```diff
@@ -110,6 +111,9 @@ def take_nd(
             return arr.take(
                 indexer, fill_value=fill_value, allow_fill=allow_fill, axis=axis
             )
+        if arr.dtype.kind in "O" and is_interval_dtype(arr):
+            # # GH46297 Interval
+            return arr.take(indexer, allow_fill=allow_fill)
```

Comment:
fill_value here comes from the na_rep argument of to_csv; is ignoring it the right behavior? Does that mean whatever the user specifies will be ignored?

Reply:
yah, this seems like a weird place to special-case